benchmarks/bench_isolation_forest_predict.py

```diff
@@ -0,0 +1,219 @@
+"""
+==========================================
+IsolationForest prediction benchmark
+==========================================
+A test of IsolationForest on classical anomaly detection datasets.
+
+The benchmark is run as follows:
+1. The dataset is randomly split into a training set and a test set, both
+assumed to contain outliers.
+2. Isolation Forest is trained on the training set fixed at 1000 samples.
+3. The test samples are scored using the trained model at:
+    - 1000, 10000, 50000 samples
+    - 10, 100, 1000 features
+    - 0.01, 0.1, 0.5 contamination
+    - 1, 2, 3, 4 n_jobs
+
+We compare the prediction time at the very end.
+
+Here are instructions for running this benchmark:
+
+1. Build this PR and run:
```

Comment:
When we merge this PR wouldn't following these instructions get kind of confusing?

Reply:
I don't follow here. What would be confusing about it? I tried to make the paths agnostic to operating system.

Reply:
I meant this PR will be closed and probably the branch associated with it deleted. So how would we actually build it?

Reply:
Ah I see. Yeah that's true. I can reword it or remove the benchmark altogether. 

I think these sort of benchmarks are useful for the developers since I can see us improving runtime even more with a more elegant thread based soln. So I can keep it. 

What do we think the policy is here?

Reply:
I tried addressing your comment here: [4adb0af](https://github.com/scikit-learn/scikit-learn/pull/28622/commits/4adb0afeacc174b8bc3e8d8b8af59f9505123784)